src/app/portlets/dot-edit-page/layout/components/dot-theme-selector/dot-theme-selector.component.ts

```diff
@@ -0,0 +1,115 @@
+import { Component, OnInit, Output, EventEmitter, Input, ViewChild, ElementRef } from '@angular/core';
+import { DotTheme } from '../../../shared/models/dot-theme.model';
+import { DotMessageService } from '../../../../../api/services/dot-messages-service';
+import { PaginatorService } from '../../../../../api/services/paginator/paginator.service';
+import { DataGrid, LazyLoadEvent } from 'primeng/primeng';
+import { Observable } from 'rxjs/Observable';
+import { Site } from 'dotcms-js/dotcms-js';
+
+/**
+ * The DotThemeSelectorComponent is modal that
+ * show the themes of the available hosts
+ * @export
+ * @class DotThemeSelectorComponent
+ */
+@Component({
+    providers: [PaginatorService],
+    selector: 'dot-theme-selector',
+    templateUrl: './dot-theme-selector.component.html',
+    styleUrls: ['./dot-theme-selector.component.scss']
+})
+export class DotThemeSelectorComponent implements OnInit {
+    themes: DotTheme[];
+    @Input() value: DotTheme;
+    @Output() selected = new EventEmitter<DotTheme>();
+    @Output() close = new EventEmitter<boolean>();
+    @ViewChild('searchInput') searchInput: ElementRef;
+    @ViewChild('dataGrid') datagrid: DataGrid;
+
+    current: DotTheme;
+
+    constructor(public dotMessageService: DotMessageService, public paginatorService: PaginatorService) {}
+
+    ngOnInit() {
+        this.dotMessageService
+            .getMessages(['editpage.layout.theme.header', 'editpage.layout.theme.search', 'dot.common.apply', 'dot.common.cancel'])
+            .subscribe();
+        this.paginatorService.url = 'v1/themes';
+        this.paginatorService.paginationPerPage = 8;
+        this.current = this.value;
```

Comment:
why do we need 'current' and 'value'?

Reply:
`value`, comes as input, `current` define the user selection while in the component, initially both values are the same, `value` acts as initial value as well.